localstack/services/awslambda/provider.py

```diff
@@ -1607,93 +1659,91 @@ def list_function_url_configs(
-                function_name, context.account_id, context.region
-            )
-            raise ResourceNotFoundException(f"Function not found: {fn_arn}", Type="User")
+        # validate qualifier
+        if qualifier is not None:
+            self._validate_qualifier_expression(qualifier)
+            if qualifier == "$LATEST":
+                raise InvalidParameterValueException(
+                    "We currently do not support adding policies for $LATEST.", Type="User"
+                )
 
-        resolved_qualifier = request.get("Qualifier", "$LATEST")
+        resolved_fn = self._get_function(function_name, context.account_id, context.region)
 
-        resource = api_utils.unqualified_lambda_arn(
-            function_name, context.account_id, context.region
-        )
-        if api_utils.qualifier_is_alias(resolved_qualifier):
-            if resolved_qualifier not in resolved_fn.aliases:
-                raise ResourceNotFoundException("Where Alias???", Type="User")  # TODO: test
-            resource = api_utils.qualified_lambda_arn(
-                function_name, resolved_qualifier, context.account_id, context.region
-            )
-        elif api_utils.qualifier_is_version(resolved_qualifier):
-            if resolved_qualifier not in resolved_fn.versions:
-                raise ResourceNotFoundException("Where Version???", Type="User")  # TODO: test
-            resource = api_utils.qualified_lambda_arn(
-                function_name, resolved_qualifier, context.account_id, context.region
-            )
-        elif resolved_qualifier != "$LATEST":
-            raise ResourceNotFoundException("Wrong format for qualifier?")
-        # TODO: is there a difference in the resulting policy when adding $LATEST manually?
+        resolved_qualifier, fn_arn = self._resolve_fn_qualifier(resolved_fn, qualifier)
 
         # check for an already existing policy and any conflicts in existing statements
         existing_policy = resolved_fn.permissions.get(resolved_qualifier)
         if existing_policy:
-            if request["StatementId"] in [s["Sid"] for s in existing_policy.policy.Statement]:
-                # TODO: is this unique just in the policy or across all policies in region/account/function (?)
-                raise ResourceConflictException("Double Statement!")
+            revision_id = request.get("RevisionId")
+            if revision_id and existing_policy.revision_id != revision_id:
+                raise PreconditionFailedException(
+                    "The Revision Id provided does not match the latest Revision Id. Call the GetFunction/GetAlias API to retrieve the latest Revision Id",
+                    Type="User",
+                )
+            request_sid = request["StatementId"]
+            if request_sid in [s["Sid"] for s in existing_policy.policy.Statement]:
+                # function version scope: statement id needs to be unique per function version
```

Comment:
sure it's per "function version"? We're using a very well defined concept for this internally and I'm not sure if that applies :thinking: 

Does it also have to be unique for aliases?

Reply:
Added extra tests and clarified comments:

```
uniqueness scope: statement id needs to be unique per qualified function ($LATEST, version, or alias)
Counterexample: the same sid can exist within $LATEST, version, and alias
```